### PR TITLE
feat: self delete team settings [AR-1000]

### PR DESCRIPTION
**Pull request by gongracr**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1000" title="AR-1000" target="_blank">StoryAR-1000</a>  Self-deleting message restrictions from TM
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
- Added needed dialog to inform user when self deleting messages team settings flag has changed, and in case it has been enforced timeout team setting has been enabled, how long now the messages will be displayed.
- Added injected usecases, `ObserveSelfDeletingMessagesUseCase` and `MarkSelfDeletingMessagesChangeAsNotifiedUseCase` to observe for changes related with the self deleting messages flag and persisting the notified flag in memory to avoid recurrent dialogs.

### Dependencies (Optional)
Needs release with:

- https://github.com/wireapp/kalium/pull/1631

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution - WIP

#### How to Test

Log into a user account that belongs to a user where team settings for self deleting messages can be modified. Then modify these and check that the dialog is displayed.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt [hunk 1]
```diff
@@ -30,29 +30,73 @@ import com.wire.android.ui.common.WireDialog
...
+    return when (rawTimeoutInSeconds.toDuration(DurationUnit.SECONDS)) {
+        timeout10Seconds -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_seconds)
+        timeout5Minutes -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_minutes)
+        timeout1Hour -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_hour)
+        timeout1Day -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_day)
+        timeout1Week -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_one_week)
+        timeout4Weeks -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled_enforced_timeout_four_weeks)
+        else -> stringResource(id = R.string.self_deleting_messages_team_setting_enabled)
     }
 }
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/HomeDialogs.kt:99]: All this is already implemented in SelfDeletionDuration, however now I see I forgot to make it work with localisations, so it needs a small refactor

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 2]
```diff
@@ -113,6 +118,29 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         }
     }
 
+    private suspend fun setSelfDeletedMessagesFeatureFlag() {
+        viewModelScope.launch {
+            observeSelfDeletingMessages().collect { selfDeletingMessagesStatus ->
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt:125]: It states to setSelfDeletedMessagesFeatureFlag, but in the end it observes self deleting messages which are emitting selfDeletingMessagesStatus 🤔

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/3/) **failed**.

**Review by Garzas — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1654/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6278417)(6278417)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6278417)(6278417)-staging-compat.apk)

**Review by borichellow — APPROVED** [review]

**Comment by gongracr** [comment]
Closing this PR in favor of https://github.com/wireapp/wire-android-reloaded/pull/1718

**Pull request closed without merging**